tests/MongoDB.Driver.Core.Tests/Core/Operations/EndTransactionOperationTests.cs

```diff
@@ -117,22 +125,49 @@ public void CreateCommand_should_return_expected_result_when_write_concern_is_se
             result.Should().Be("{ fakeOperation : 1 }");
         }
 
+        [Fact]
+        public void RecoveryToken_should_return_expected_result()
+        {
+            var recoveryToken = BsonDocument.Parse("{ hp: 42} ");
+            var subject = CreateSubject(recoveryToken: recoveryToken);
+
+            var result = subject.RecoveryToken;
+
+            result.Should().BeSameAs(recoveryToken);
+        }
+
         // private methods
         private EndTransactionOperation CreateSubject(
+            BsonDocument recoveryToken = null,
             WriteConcern writeConcern = null,
             MessageEncoderSettings messageEncoderSettings = null)
         {
             writeConcern = writeConcern ?? new WriteConcern();
-            return new FakeEndTransactionOperation(writeConcern)
+            if (recoveryToken == null)
+            {
+                return new FakeEndTransactionOperation(writeConcern)
+                {
+                    MessageEncoderSettings = messageEncoderSettings
+                };
+            }
+            else
             {
-                MessageEncoderSettings = messageEncoderSettings
-            };
+                return new FakeEndTransactionOperation(recoveryToken, writeConcern)
+                {
+                    MessageEncoderSettings = messageEncoderSettings
+                };
+            }
         }
 
         // nested types
         private class FakeEndTransactionOperation : EndTransactionOperation
         {
             // public constructors
+            public FakeEndTransactionOperation(BsonDocument recoveryToken, WriteConcern writeConcern)
+                : base(recoveryToken, writeConcern)
+            {
+            }
+
```

Comment:
No need to keep the original constructor. This is a private class.

Reply:
Fixed